contracts/MyToken.sol

```diff
@@ -0,0 +1,57 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.9;
+
+import "openzeppelin-solidity/contracts/token/ERC20/ERC20.sol";
+import "openzeppelin-solidity/contracts/access/Ownable.sol";
+import "openzeppelin-solidity/contracts/security/Pausable.sol";
+
+contract MyToken is ERC20,Pausable,Ownable {
+
+    mapping (address => uint256) internal _balances;
+
+    constructor() ERC20("MyToken", "MK") {
+        uint256 balance =  1_000_000 * 10 ** decimals();
+        _mint(msg.sender,balance);  
+        _balances[msg.sender] = balance; 
```

Comment:
this action is not required. delete it. 
_mint already puts this amount on the balance, you just rewrite it